servicetalk-tcp-netty-internal/src/main/java/io/servicetalk/tcp/netty/internal/EarlyConnectionContext.java

```diff
@@ -57,12 +58,12 @@ public String connectionId() {
 
     @Override
     public SocketAddress localAddress() {
-        return channel.localAddress();
+        return fromNettyAddress(channel.localAddress());
```

Comment:
Does it make sense to cache the result, or to save it as a field on initialization? The latter may be risky if we're constructing these before connections are established, so caching may be the only option.

I'm not strongly opinionated, especially considering how infrequently domain sockets are used, so feel free to ignore.